controllers/users/checkAllowAnonDM.js

```diff
@@ -5,9 +5,12 @@ const Getstream = require('../../vendor/getstream');
 
 module.exports = async (req, res) => {
   const {userId: self_user_id} = req;
-  let {post_id} = req.params;
+  let {source, post_id, comment_id} = req.query;
 
-  const target_user_id = await Getstream.feed.getUserIdFromSource(res, 'post', {post_id});
+  const target_user_id = await Getstream.feed.getUserIdFromSource(res, source, {
+    post_id,
+    comment_id
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces the `source` and `comment_id` query parameters, but there is no validation of these parameters before they are used. This could lead to potential security issues such as SQL injection or unexpected behavior if invalid values are provided. It's recommended to validate these parameters before using them.

```diff
+  const schema = Joi.object({
+    source: Joi.string().valid('post', 'comment').required(),
+    post_id: Joi.when('source', { is: 'post', then: Joi.number().integer().required(), otherwise: Joi.forbidden() }),
+    comment_id: Joi.when('source', { is: 'comment', then: Joi.number().integer().required(), otherwise: Joi.forbidden() })
+  });
+
+  const { error, value } = schema.validate({ source, post_id, comment_id });
+  if (error) {
+    return res.status(400).send({ message: error.details[0].message });
+  }
+  
+  source = value.source;
+  post_id = value.post_id;
+  comment_id = value.comment_id;

   const target_user_id = await Getstream.feed.getUserIdFromSource(res, source, {
     post_id,
     comment_id
   });
```
This will ensure that `source` is either 'post' or 'comment', and depending on the value of `source`, either `post_id` or `comment_id` is required and must be an integer. If the validation fails, it will return a 400 status with a descriptive error message.


<!-- This is an auto-generated comment by OSS CodeRabbit -->